uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeaveRequestTest.php

```diff
@@ -277,6 +287,177 @@ public function testGetReturnsOnlyPublicHolidayLeaveRequestsIfThePublicHolidayIs
     $this->assertEquals($publicHoliday->date, $result['values'][0]['from_date']);
   }
 
+  public function testGetCanReturnALeaveRequestWhichOverlapsAContractWithoutEndDate() {
+    $leaveRequestStatuses = array_flip(LeaveRequest::buildOptions('status_id'));
+
+    HRJobContractFabricator::fabricate([
+      'contact_id' => 1
+    ],
+    [
+      'period_start_date' => '2016-01-01'
+    ]);
+
+    //This leave request is before the contract start date and will not be returned
+    LeaveRequestFabricator::fabricate([
+      'contact_id' => 1,
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2015-12-30'),
+      'to_date' => CRM_Utils_Date::processDate('2015-12-31'),
+      'status_id' => $leaveRequestStatuses['Waiting Approval']
+    ], true);
+
+    //This will be returned as it is after the contract start date
+    $leaveRequest2 = LeaveRequestFabricator::fabricate([
+      'contact_id' => 1,
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2017-12-30'),
+      'to_date' => CRM_Utils_Date::processDate('2017-12-31'),
+      'status_id' => $leaveRequestStatuses['Waiting Approval']
+    ], true);
+
+    $result = civicrm_api3('LeaveRequest', 'get');
+    $this->assertCount(1, $result['values']);
+    $this->assertNotEmpty($result['values'][$leaveRequest2->id]);
+  }
+
+  public function testGetCanReturnALeaveRequestWhichOverlapsAContractWithEndDate() {
```

Comment:
Can we also have a test that returns multiple leave requests overlapping a contract rather than just returning a single leave request that overlaps the contract dates?